Upgrade bveOXD/OXD script

sajanrajdev opened the pull request:
Closes issue #554 

Run with:
```
brownie run scripts/issue/554/upgrade_bveOXD_LP.py --netowork ftm-main-fork
```

Run simulation with:
```
brownie run scripts/issue/554/upgrade_bveOXD_LP.py main true --netowork ftm-main-fork
```

btcookies commented on scripts/issue/554/upgrade_bveOXD_LP.py:
```diff
@@ -0,0 +1,48 @@
+            strat.harvest({"from": safe.account})
+
+    # Execute upgrade
+    proxyAdmin = interface.IProxyAdmin(DEV_PROXY, owner=safe.account)
```
should snake_case for consistence

sajanrajdev replied:
Good call, fixed

btcookies commented on scripts/issue/554/upgrade_bveOXD_LP.py:
```diff
@@ -0,0 +1,48 @@
+    proxyAdmin.upgrade(strat.address, NEW_LOGIC)
+
+    # Confirm storage variables
+    for attr in strat_contract.signatures:
```
should we make this a function somewhere? like a generalized check_storage(contract)? and also think we should fail if one of the attributes isn't equal so maybe have something like following

```
is_storage_wrong = False
for attr in attrs:
   try:
      whatever
   except:
      is_storage_wrong = True

if is_storage_wrong:
    fail 
```

btcookies replied:
Hmm should we at least log which ones pass and which ones fail so we can at least manually visually confirm everything looks good?

shuklaayush replied:
Btw, this isn't an exhaustive check since it only verifies public value-type storage variables and public view functions

sajanrajdev replied:
That's right, it has limitations. Not sure if this approach could be generalized efficiently.

shuklaayush reviewed (approved):
LGTM

petrovska-petro commented on scripts/issue/554/upgrade_bveOXD_LP.py:
```diff
@@ -38,7 +38,9 @@ def main(simulation="false"):
             assert attributes[attr] == getattr(strat_contract, attr).call()
-        except:
+        # Should fail if assertion is wrong
+        except KeyError:
```
catching only this exception for me locally seems to jump out in other type of exception `AssertionError`.

maybe should be instead `except (AssertionError, KeyError) as e:` or just check for *attr* which are in the *attributes* variable

sajanrajdev replied:
Interesting, I didn't get any assertion errors. The idea was for assertion errors to jump out, they would mean that either a view variable's value changed and this change should be accounted or that a new variable was introduced. In this case, neither of these happened and I don't see why this would be the case. 

Do you know which attribute caused the assertion error?

petrovska-petro replied:
for the record was **balanceOfRewards** discussed in dm

sajanrajdev commented:
Changed script to fetch contract from the explorer so that it retrives the latest instance (instead of the stored locally) with all the attributes. This showed the balanceOfRewards didn't match before and after running. This is due to rewards increasing slightly over the time difference. Added a check to ensure that these balances indeed only increase.

Additionally, catching for assertion errors was introduced.

CC @petrovska-petro

petrovska-petro reviewed (approved):
LGTM! ✅
